src/java/org/apache/cassandra/config/GuardrailsOptions.java

```diff
@@ -194,6 +197,31 @@ public int getMaterializedViewsPerTableWarnThreshold()
         return config.materialized_views_per_table_warn_threshold;
     }
 
+    @Override
```

Comment:
Here you have 2 getters but only one setter for both. This is usually a pain bc it forces you to do a read before write when you want to set only one. It's a nit but I thought I'd raise it.

Reply:
This single setter approach is used by all thresholds. It was discussed during [CASSANDRA-17147](https://github.com/apache/cassandra/pull/1321#), see[this comment](https://github.com/apache/cassandra/pull/1321#discussion_r760697524). The idea is that the validation of each warn/fail thresholds is tied to the other, so in the end if you want to change one you need to know the other.